kroxylicious-kms-provider-hashicorp-vault/src/test/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultKmsIT.java

```diff
@@ -45,30 +45,28 @@ class VaultKmsIT {
 
     private static final String HASHICORP_VAULT = "hashicorp/vault:1.15";
     @SuppressWarnings("rawtypes")
-    private VaultContainer vaultContainer;
-    private VaultKms service;
+    private static VaultContainer vaultContainer;
+    private static VaultKms service;
 
-    @BeforeEach
-    @SuppressWarnings("resource")
-    void beforeEach() {
+    @BeforeAll
```

Comment:
What was the motivation for the lifecycle change?  I tend to use beforeEach for the certainty that there's no contamination between tests, unless the overheads are just too high.  Docker containers spin up really quickly, so in this case I went beforeEach.  

I'm not against the change, just curious about your thought process.

Reply:
The tests take ~2seconds each for me, earlier I had the branch running the 7 existing tests against all the TLS configurations so it was taking quite a long time. I didn't want to add another minute to the suite. But since refactoring to have a smaller TLS suite I think it's better to go back to the cleaner way.

In general I worry that the IT suite is steadily pushing the build time up and that the time to hear back from sonar is getting quite big. Maybe we should at making it parallelisable so we can have 2 surefire forks.

Reply:
I'm not sure there is much to win from parallelisation in the Git Hub action runners as they are iirc 2core. 

I'm tempted to say we change the trigger on the sonar run so it doesn't wait for a successful build. Or the more drastic open we restructure the build pipeline so that we have what I would term a commit build run which then triggers sonar and the integration tests in parallel. By a commit build I mean compile and unit test run, plus associated formatting checks/fixes.

Reply:
Have refactored both ITs back to creating a vault per test. The vault kms tests take ~30seconds so something to keep an eye on.